### PR TITLE
clippy: fix clippy lints for 1.83

### PR DESCRIPTION
**Pull request by muzarski**
- derive(FromPrimitive) does not work with new clippy. Need to implement it manually We can maybe revert it in the future. However, this change is not a big deal.
- Converted all manually created c-strings to c"" literals
- Removed some unnecessary return statements detected by clippy

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have implemented Rust unit tests for the features/changes introduced.
- [ ] I have enabled appropriate tests in `.github/workflows/build.yml` in `gtest_filter`.
- [ ] I have enabled appropriate tests in `.github/workflows/cassandra.yml` in `gtest_filter`.

### CODE REVIEW

File: scylla-rust-wrapper/src/cluster.rs [hunk 1]
```diff
@@ -1070,6 +1070,8 @@ mod tests {
                     );
 
                     // empty strings
+                    // Allow it, since c"" somehow clashes with #[ntest] proc macro...
+                    #[allow(clippy::manual_c_str_literals)]
```
Comment (Lorak-mmk) [inline scylla-rust-wrapper/src/cluster.rs:1074]: Eh, why is ntest so problematic...

**Review by Lorak-mmk — APPROVED** [review]
> Converted all manually created c-strings to c"" literals

Love it! But I'm surprised there is that few of them, I thought there was much more. Isn't there more in error stringifying function etc?

**Comment by muzarski** [comment]
> > Converted all manually created c-strings to c"" literals
> 
> Isn't there more in error stringifying function etc?

You are right. I'm not sure why clippy did not detect them. I'll fix those in a separate commit.

**Comment by muzarski** [comment]
v2: replaced all occurrences of `"...\0"` with `c"..."` string literals

File: scylla-rust-wrapper/src/cluster.rs (hunk 1 above)
Comment (muzarski) [reply scylla-rust-wrapper/src/cluster.rs:1074]: I opened an issue: https://github.com/becheran/ntest/issues/30

**Review by wprzytula — APPROVED** [review]

**Review by Lorak-mmk — APPROVED** [review]

**Pull request merged**